packages/ui/src/elements/RenderServerComponent/index.tsx

```diff
@@ -50,6 +53,18 @@ export const RenderServerComponent: RenderServerComponentFn = ({
       ...(isRSC ? serverProps : {}),
     })
 
+    if (HOC) {
+      return (
+        <WithHOC
```

Comment:
I would call WithHOC as a function instead of rendering it - that way, we don't send the `WithHOC` logic to the client